### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by kpp**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -53,12 +53,11 @@
 //! ```compile_fail
 //!# use compact_arena::mk_tiny_arena;
 //! let idx = { mk_tiny_arena!(arena, 1); arena.add(1usize)) };
-//! assert!(core::mem::size_of_val(&a) == 4);
```
Comment (llogiq) [inline src/lib.rs]: Please don't remove this, change the value to 2 instead.

**Comment by llogiq** [comment]
Thanks for the PR! Looks good to me apart from one small nit.

File: src/lib.rs (hunk 1 above)
Comment (kpp) [reply src/lib.rs]: there is no `a` in the scope.

Comment (llogiq) [reply src/lib.rs]: Sorry, my bad, that should be `idx`. Please try to remove the `compile-fail` to ensure you get a lifetime error.

Comment (kpp) [reply src/lib.rs]: Anyway why should it be `idx`?

Comment (kpp) [reply src/lib.rs]: Here is how it looks:

```
---- src/lib.rs -  (line 53) stdout ----
error[E0597]: `tag` does not live long enough
 --> src/lib.rs:55:13
  |
5 | let idx = { mk_tiny_arena!(arena); arena.add(1usize) };
  |     ---     ^^^^^^^^^^^^^^^^^^^^^^                   - `tag` dropped here while still borrowed
  |     |       |
  |     |       borrowed value does not live long enough
  |     borrow later stored here
```

**Comment by kpp** [comment]
Also I found a bug:

```
//! ```compile_fail
//!# use compact_arena::mk_nano_arena;
//! mk_nano_arena!(a);
//! mk_nano_arena!(b);
//! let i = a.add(1usize);
//! let j = b.add(1usize);
//! let _ = b[i];
//! ```
```

**Comment by llogiq** [comment]
Oh dear. This seems to hinge on the fact that the arena is not `Copy`, because in that case the drop check doesn't apply.  @RalfJung can you confirm this or is there a different reason?

In any case, we better remove the uninit feature then, make that code the default and create a new nightly feature that uses `MaybeUninit`.

**Comment by RalfJung** [comment]
Uh, not sure, what is going on here?

**Comment by RalfJung** [comment]
Hm, [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e488639b5f70175fc4b09f3cbedaaff8) also gets accepted. So maybe one cannot use the lifetime of a local after all? Seems like NLL has too much control here.

**Comment by llogiq** [comment]
In that case, we best reactivate the unique type tags. It *appears* the invariant lifetime catches recursion, so having both *might* suffice. However I wouldn't want to be the one to prove this...

**Comment by llogiq** [comment]
So, this means we can no longer have `mk_arena!`, because that would mean using two of them would give us two `Tag` types in the same scope which is an error. However, [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=972bdd54d1fac1b6a3447c92c235d9ab) might work, using the previous `in_arena!` macro with a side of invariant lifetime.

**Comment by llogiq** [comment]
Otherwise we'd need to get a scope-unique identifier for the lifetime.

**Comment by kpp** [comment]
> Otherwise we'd need to get a scope-unique identifier for the Tag.

How about creating a unique ident for a Tag struct with file! line! column! ? To get struct Tag_name_line_column {}

**Comment by kpp** [comment]
I reverted that:

```
//! ```compile_fail
//!# use compact_arena::mk_nano_arena;
//! mk_nano_arena!(a);
//! mk_nano_arena!(b);
//! let i = a.add(1usize);
//! let j = b.add(1usize);
//! let _ = b[i];
//! ```
```

This should be fixed in another issue

**Comment by RalfJung** [comment]
> In that case, we best reactivate the unique type tags. It appears the invariant lifetime catches recursion, so having both might suffice. However I wouldn't want to be the one to prove this...

Well, clearly the borrow checker is doing *something* here that we don't want it to do, and without understand what that is I'd not trust any of these schemes.

**Comment by llogiq** [comment]
An interesting fact is that it appears to work when the Arena can `drop`.

**Comment by llogiq** [comment]
It may be interesting to look if we can get the correct lifetime check by adding a value with `drop` to the arena.

**Comment by llogiq** [comment]
Even more interestingly, I couldn't reproduce the error by copying your playground example into a doctest. Maybe the crate barrier does something that fails within a single crate?

**Comment by kpp** [comment]
@llogiq would you please merge this PR and create a special issue for this case?

**Comment by llogiq** [comment]
Sure!

**Pull request merged**